### PR TITLE
Sourcery Starbot ⭐ refactored jonathangjertsen/jchord

### PR DESCRIPTION
**Pull request by SourceryAI**
Thanks for starring [sourcery-ai/sourcery](/sourcery-ai/sourcery) ✨ 🌟 ✨

Here's your pull request refactoring your most popular Python repo.

If you want Sourcery to refactor all your Python repos and incoming pull requests [install our bot](/apps/sourcery-ai/installations/new).

<details>
<summary>Review changes via command line</summary>
<p>To manually merge these changes, make sure you're on the <code>master</code> branch, then run:</p>
<pre>
git fetch https://github.com/sourcery-ai-bot/jchord master
git merge --ff-only FETCH_HEAD
git reset HEAD^
</pre>
</details>

### CODE REVIEW

File: jchord/chords.py [hunk 1]
```diff
@@ -90,7 +90,8 @@ def _chord_options_triad_with_lower_note(lower_note, upper_triad, _rec):
     return [
         option
         for option in options
-        if not ((bass_degree in option and "(no{})".format(base_degree) in option))
+        if bass_degree not in option
+        or "(no{})".format(base_degree) not in option
```
Comment (SourceryAI) [inline jchord/chords.py:94]: Function \_chord\_options\_triad\_with\_lower\_note refactored with the following changes:
- Simplify logical expression

File: jchord/chords.py [hunk 2]
```diff
@@ -120,7 +121,7 @@ def semitones_to_chord_name_options(semitones: Set[int], _rec=5) -> List[str]:
 
     semitones_sorted = list(sorted(semitones))
 
-    if len(semitones) == 0:
+    if not semitones:
```
Comment (SourceryAI) [inline jchord/chords.py:124]: Function semitones\_to\_chord\_name\_options refactored with the following changes:
- Simplify sequence comparison

File: jchord/chords.py [hunk 3]
```diff
@@ -154,10 +155,7 @@ def from_semitones(cls, name: Optional[str], semitones: List[int]) -> "Chord":
         """
         if name is None:
             name_options = semitones_to_chord_name_options(semitones)
-            if name_options:
-                name = name_options[0]
-            else:
-                name = cls.UNNAMED
+            name = name_options[0] if name_options else cls.UNNAMED
```
Comment (SourceryAI) [inline jchord/chords.py:158]: Function Chord.from\_semitones refactored with the following changes:
- Replace if statement with if expression

File: jchord/chords.py [hunk 4]
```diff
@@ -201,10 +199,10 @@ def intervals(self) -> List[int]:
...
         * `Chord.from_name("major7").intervals() == [4, 3, 4]`
         """
-        intervals = []
-        for i in range(1, len(self.semitones)):
-            intervals.append(self.semitones[i] - self.semitones[i - 1])
-        return intervals
+        return [
+            self.semitones[i] - self.semitones[i - 1]
+            for i in range(1, len(self.semitones))
+        ]
```
Comment (SourceryAI) [inline jchord/chords.py:205]: Function Chord.intervals refactored with the following changes:
- Convert for loop into list comprehension
- Inline variable that is only used once

File: jchord/midi.py [hunk 5]
```diff
@@ -85,8 +85,7 @@ def _events_to_notes(events: list) -> List[PlayedNote]:
 def read_midi_file(filename: str) -> List[PlayedNote]:
     """Reads the MIDI file for the given filename and returns the corresponding list of `PlayedNote`s."""
     events = _read_midi_file_to_events(filename)
-    notes = _events_to_notes(events)
-    return notes
+    return _events_to_notes(events)
```
Comment (SourceryAI) [inline jchord/midi.py:88]: Function read\_midi\_file refactored with the following changes:
- Inline variable that is only used once

File: jchord/progressions.py [hunk 6]
```diff
@@ -107,11 +107,7 @@ def to_string(
         output = []
         prev_chord = None
         for chord in self.progression:
-            if prev_chord == chord:
-                chord_name = REPETITION_SYMBOL
-            else:
-                chord_name = chord.name
-
+            chord_name = REPETITION_SYMBOL if prev_chord == chord else chord.name
```
Comment (SourceryAI) [inline jchord/progressions.py:110]: Function ChordProgression.to\_string refactored with the following changes:
- Replace if statement with if expression

File: jchord/progressions.py [hunk 7]
```diff
@@ -151,11 +147,7 @@ def to_xlsx(self, filename: str, chords_per_row: int = 4):
         column = 1
         prev_chord = None
         for chord in self.progression:
-            if prev_chord == chord:
-                chord_name = REPETITION_SYMBOL
-            else:
-                chord_name = chord.name
-
+            chord_name = REPETITION_SYMBOL if prev_chord == chord else chord.name
```
Comment (SourceryAI) [inline jchord/progressions.py:150]: Function ChordProgression.to\_xlsx refactored with the following changes:
- Replace if statement with if expression

**Comment by jonathangjertsen** [comment]
not impressed

**Pull request closed without merging**